bespin/amazon/cloudformation.py

```diff
@@ -58,20 +60,28 @@ class UPDATE_ROLLBACK_FAILED(Status): pass
 class UPDATE_ROLLBACK_COMPLETE_CLEANUP_IN_PROGRESS(Status): pass
 class UPDATE_ROLLBACK_COMPLETE(Status): pass
 
+# REVIEW_IN_PROGRESS only valid for CreateChangeSet with ChangeSetType=CREATE
+class REVIEW_IN_PROGRESS(Status): pass
+
 for kls in [Status] + Status.__subclasses__():
     with_meta = six.add_metaclass(StatusMeta)(kls)
     locals()[kls.__name__] = with_meta
     Status.statuses[kls.__name__] = with_meta
 
+##BOTO3 TODO: refactor to use boto3 resources
 class Cloudformation(AmazonMixin):
-    def __init__(self, stack_name, region="ap-southeast-2"):
+    def __init__(self, stack_name, region="ap-southeast-2", session=None):
         self.region = region
         self.stack_name = stack_name
+        self.session = session
+        if self.session is None:
+            self.session = boto3.session.Session(region_name=self.region)
+        assert self.session.region_name == self.region
```

Comment:
I'm not a fan of asserts in code.

I'd rather we make session a memoized property than allow it to be passed in

    @hp.memoized_property
    def session(self):
        return boto3.session.Session(region_name=self.region)